File: lib/very_easy/temperature_converter.dart
double toCelsius(double fahrenheit) {
  return (fahrenheit - 32) * 5 / 9;
}

double toFahrenheit(double celsius) {
  return (celsius * 1.8) + 32;
}

double toKelvin(double celsius) {
  return celsius + 273.15;
}

List<double> convertTemperature(double celsius) {
  return [toKelvin(celsius), toFahrenheit(celsius)];
}

bool isDigit(int char) {
  return char >= 48 && char <= 57;
}

bool isTemperatureChar(String char) {
  switch (char) {
    case '-':
    case '.':
      return true;
    default:
      return false;
  }
}

String temperatureConverter(String deg) {
  final buffer = StringBuffer();
  for (int i = 0; i < deg.length; i++) {
    if (isDigit(deg.codeUnitAt(i)) || isTemperatureChar(deg[i])) {
      buffer.write(deg[i]);
    } else {
      break;
    }
  }
  double temp = double.parse(buffer.toString());
  if (deg.endsWith("°C")) {
    double temperature = toFahrenheit(temp);
    return "${temperature.round()}°F";
  }

  if (deg.endsWith("°F")) {
    double temperature = toCelsius(temp);
    return "${temperature.round()}°C";
  }
  return "Error";
}

File: src/very_easy/temperature-converter.ts
export const toCelsius = (temperature: number): number =>
    ((temperature - 32) * 5) / 9;
export const toFahrenheit = (temperature: number): number =>
    temperature * 1.8 + 32;
export const toKelvin = (temperature: number) => temperature + 273.15;
export const convertTemperature = (celsius: number): number[] => [
    toKelvin(celsius),
    toFahrenheit(celsius),
];

export const temperatureConverter = (deg: string): string => {
    let temp = "";
    for (const char of deg) {
        if (/[0-9.]/.test(char) || char === "-") {
            temp += char;
        }
    }
    const temperature = Number(temp);
    if (deg.endsWith("°F")) return `${Math.round(toCelsius(temperature))}°C`;
    if (deg.endsWith("°C")) return `${Math.round(toFahrenheit(temperature))}°F`;
    return "Error";
};
